qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,431 @@
+        # validation
+        self._check_operator_ansatz(operator)
+
+        initial_point = validate_initial_point(self.initial_point, self.ansatz)
+
+        bounds = validate_bounds(self.ansatz)
+
+        # We need to handle the array entries being zero or Optional i.e. having value None
+        if aux_operators:
+            zero_op = PauliSumOp.from_list([("I" * self.ansatz.num_qubits, 0)])
+
+            # Convert the None and zero values when aux_operators is a list.
+            # Drop None and convert zero values when aux_operators is a dict.
+            if isinstance(aux_operators, list):
+                key_op_iterator = enumerate(aux_operators)
+                converted = [zero_op] * len(aux_operators)
+            else:
+                key_op_iterator = aux_operators.items()
+                converted = {}
+            for key, op in key_op_iterator:
+                if op is not None:
+                    converted[key] = zero_op if op == 0 else op
+
+            aux_operators = converted
+
+        else:
+            aux_operators = None
+
+        if self.betas is None:
+            if isinstance(operator, PauliSumOp):
+                upper_bound = abs(operator.coeff) * sum(
+                    abs(operation.coeff) for operation in operator
+                )
+                self.betas = [upper_bound * 10] * (self.k)
+                logger.info("beta autoevaluated to %s", self.betas[0])
+            else:
+                raise NotImplementedError(
+                    r"Beta autoevaluation is only supported for operators"
+                    f"of type PauliSumOp, found {type(operator)}."
+                )
+
+        result = self._build_vqd_result()
+
+        if aux_operators is not None:
+            aux_values = []
+
+        for step in range(1, self.k + 1):
+
+            self._eval_count = 0
+            energy_evaluation = self._get_evaluate_energy(
+                step, operator, prev_states=result.optimal_parameters
+            )
+
+            start_time = time()
+
+            # TODO: add gradient support after FidelityGradients are implemented
+            if callable(self.optimizer):
+                opt_result = self.optimizer(  # pylint: disable=not-callable
+                    fun=energy_evaluation, x0=initial_point, bounds=bounds
+                )
+            else:
+                opt_result = self.optimizer.minimize(
+                    fun=energy_evaluation, x0=initial_point, bounds=bounds
+                )
+
+            eval_time = time() - start_time
+
+            self._update_vqd_result(result, opt_result, eval_time, self.ansatz)
+
+            if aux_operators is not None:
+                aux_value = estimate_observables(
+                    self.estimator, self.ansatz, aux_operators, result.optimal_point[-1]
+                )
+                aux_values.append(aux_value)
+
+            if step == 1:
+
```

Comment:
This blank line seems odd after the if (not one after else) - pretty minor comment!